core/api/src/main/java/com/blazebit/persistence/spi/CriteriaBuilderConfigurationContributor.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2014 - 2021 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.spi;
+
+/**
+ * A bootstrap process hook for contributing settings to the {@link CriteriaBuilderConfiguration}.
+ *
+ * Implementations are instantiated via {@link java.util.ServiceLoader}.
+ *
+ * @author Jan-Willem Gmelig Meyling
+ * @since 1.6.2
+ */
+public interface CriteriaBuilderConfigurationContributor {
```

Comment:
IMO this should also include a `int ordinal()` method with a description like `The ordinal is used to determine ordering of the contributors. The range 0-500 is reserved for internal uses. 500 - 1000 is reserved for libraries and 1000+ is for user provided contributors.`

Reply:
What about a `Priority` annotation (like JAX-RS filters). I do understand the wish for deterministic ordering on one hand, and control over the ordering on the other hand, but I am not sure whether I like it to bleed into the SPI itself.

Reply:
Hmm, I guess that would be fine as long as we only have an optional dependency on the javax.annotation-api artifact.

Reply:
I was thinking to introduce our own annotation, however, I can also use the one from` javax.annotation-api` and its not unreasonable for users to expect that to just work.

I would need to extract the value through reflection though.

Reply:
A pro for an internal annotation would be that if we actually were to use the SPI ourselves, we would endup with a compile time dependency on `javax.annotation-api` anyway.

Reply:
Hmm, I guess a custom annotation in core-api is probably a better choice indeed. It would be good to still also support `@Priority` from javax.annotation-api though.

Reply:
See `CriteriaBuilderConfigurationContributorComparator`.